### PR TITLE
fix(ui): Include application name in status badge

### LINKED ISSUE #14162: Include application name in the badge SVG
# Summary

I'd like to propose to optionally include the project name in the badge SVG.

# Motivation

Similar to [GitHub badges](https://docs.github.com/en/actions/monitoring-and-troubleshooting-workflows/adding-a-workflow-status-badge), it would be convenient if the Argo badges also included the project name (optionally). This would let the badge be displayed without additional labels

# Proposal

Similar to the optional revision information (`?revision=true`) the name could be displayed with (`?displayName=true`, since `name` is already used)

### PR DESCRIPTION
**Pull request by sonamkshenoy**
Resolves [#14162](https://github.com/argoproj/argo-cd/issues/14162)

Screenshot 2024-02-07 at 11 20 45 PM         
     
Screenshot 2024-02-07 at 11 32 06 PM

### CODE REVIEW

**Review by todaywasawesome — CHANGES_REQUESTED** [review]
Can we do this as an alternate version for people that don't want the app name? And then update the docs https://argo-cd.readthedocs.io/en/stable/user-guide/status-badge/

**Comment by crenshaw-dev** [comment]
+1 to @todaywasawesome. Let's add something like a `showAppName` param to enable this presentation.

**Comment by sonamkshenoy** [comment]
Thanks for the feedback!
I have made the suggested changes.
Could you please have another look. Thanks!

**Review by todaywasawesome — APPROVED** [review]
LGTM

File: docs/user-guide/status-badge.md [hunk 1]
```diff
@@ -9,7 +9,11 @@ To show this badge, use the following URL format `${argoCdBaseUrl}/api/badge?nam
...
+2. Scroll down to 'Status Badge' section.
+3. Select required template such as URL, Markdown etc.
 for the status image URL in markdown, html, etc are available .
-1. Copy the text and paste it into your README or website.
\ No newline at end of file
+4. Copy the text and paste it into your README or website.
+
+To be able to see the application name in the application status badge use the `?showAppName=true` query parameter.   
+For instance, `${argoCdBaseUrl}/api/badge?name=${appName}&showAppName=true`.   
+Naturally, to remove the application name from the badge, remove the query parameter from the URL or set it to `false`.
```
[the remark this answers is not in the document]
Comment (sonamkshenoy) [reply docs/user-guide/status-badge.md]: Hey @todaywasawesome, I'm really sorry for missing this. For some weird reason, I didn't receive notifications for any of the above review comments. I'm so sorry you had to make the changes. Sincere apologies again for the inconvenience!

**Review by pasha-codefresh — APPROVED** [review]
LGTM

**Pull request merged**